frontend/app-development/layout/SettingsModalButton/SettingsModal/utils/tabUtils/policyTabUtils/policyTabUtils.ts

```diff
@@ -0,0 +1,92 @@
+ * @returns {PolicyAction[]} - The merged policy actions.
+ */
+export const mergeActionsFromPolicyWithActionOptions = (
+  rules: PolicyRule[],
+  actions: PolicyAction[],
+): PolicyAction[] => {
+  const existingActionIds = getExistingActionIds(actions);
+  const copiedActions = deepCopy(actions);
+
+  rules.forEach((rule) => {
+    rule.actions.forEach((actionString) => {
+      if (!existingActionIds.includes(actionString)) {
+        const newAction: PolicyAction = createNewActionFromActionString(actionString);
+        copiedActions.push(newAction);
+        existingActionIds.push(actionString);
+      }
+    });
+  });
+
+  return copiedActions;
+};
+
+export const getExistingActionIds = (actions: PolicyAction[]): string[] =>
+  actions.map((action) => action.actionId);
+
+export const createNewActionFromActionString = (actionString: string): PolicyAction => ({
+  actionId: actionString,
+  actionTitle: actionString,
+  actionDescription: null,
+});
+
+/**
+ * Merges subjects from policy rules with existing subject options.
+ *
+ * @param {PolicyRule[]} rules - The policy rules containing subjects to be merged.
+ * @param {PolicySubject[]} subjects - The existing policy subjects to merge with.
+ *
+ * @returns {PolicySubject[]} - The merged policy subjects.
+ */
+export const mergeSubjectsFromPolicyWithSubjectOptions = (
+  rules: PolicyRule[],
+  subjects: PolicySubject[],
+): PolicySubject[] => {
+  const existingSubjectIds = getExistingSubjectIds(subjects);
+  const copiedSubjects = deepCopy(subjects);
+
+  rules.forEach((rule) => {
+    rule.subject.forEach((subjectString) => {
+      const subjectId = convertSubjectStringToSubjectId(subjectString);
+
+      if (!existingSubjectIds.includes(subjectId)) {
+        const newSubject: PolicySubject = createNewSubjectFromSubjectString(subjectString);
+        copiedSubjects.push(newSubject);
+        existingSubjectIds.push(subjectId);
+      }
+    });
+  });
+
+  return copiedSubjects;
+};
+
+export const getExistingSubjectIds = (subjects: PolicySubject[]): string[] =>
+  subjects.map((subject) => subject.subjectId);
+
+export const convertSubjectStringToSubjectId = (subjectString: string): string => {
+  const subjectStringAsArray: string[] = subjectString.split(':');
+  return subjectStringAsArray[subjectStringAsArray.length - 1]; // The final element is the ID
+};
+
+export const createNewSubjectFromSubjectString = (subjectString: string): PolicySubject => {
+  const subjectId: string = convertSubjectStringToSubjectId(subjectString);
+  return {
+    subjectId: subjectId,
+    subjectTitle: subjectId,
+    subjectSource: convertSubjectStringToSubjectSource(subjectString),
+    subjectDescription: '',
+  };
+};
+
+export const convertSubjectStringToSubjectSource = (subjectString: string): string => {
+  const subjectStringAsArray: string[] = subjectString.split(':');
+  return subjectStringAsArray.slice(1, subjectStringAsArray.length - 1).join(':'); // Starting at 1 to remove 'urn'
+};
```

Comment:
Mulig jeg missforstod koden du hadde, men det du ønsker å oppnå er vel bare å fjerne urn:, men beholde resten av stringen?
```suggestion
export const convertSubjectStringToSubjectSource = (subjectString: string): string => {
   return subjectString.replace(/^urn:/, '');
};
```

Reply:
Ja, er egentlig det man ønsker. Men, som vi oppdaget i går så kan folk skrive feil om de bruker løsningen gjennom sin egen editor. Så jeg tenkte at å sjekke på en string verdi ikke var det lureste. Hva om brukeren skriver? ```urm``` med et uhell? Så tenkte det var safere å fjerne første element før ```:```

Reply:
ah, smart tenkt - kan det løses sånn som dette? 😊 
```suggestion
export const convertSubjectStringToSubjectSource = (subjectString: string): string => {
const firstColonIndex = inputString.indexOf(':');
const modifiedString = inputString.slice(firstColonIndex + 1);
};
```

Reply:
Ja, men trenger index av siste også for å fjerne ID 😄